### DISCUSSION TITLE
How to support it in the Open JDK 11? How to update the code to support it?

### DISCUSSION THREAD

**Discussion by mendickxiao** [DependencyTrack/dependency-track#859] [Q&A]
Please use Java 8. That is what the official Docker containers use. Java 11 has breaking changes - this is one of them.

_Originally posted by @stevespringett in https://github.com/DependencyTrack/dependency-track/issues/617#issuecomment-598842310_

**Comment by stevespringett** [comment]
Dependency-Track runs fine on Java 11. However, you have to load the database drivers in the classpath yourself. Also note that the only Java distribution officially supported is AdoptOpenJDK, not the upstream OpenJDK releases built by RedHat. Theoretically, they should work fine, but they are not tested or supported.

Also note, extensive testing reveals that Dependency-Track v3.x is noticeably slower on Java 11 than on Java 8, especially under heavy load when multiple concurrent processes where happening and GC was taking place. Running Dependency-Track on Java 11 is not recommended.

AdoptOpenJDK is fully supported until 2026 (including security backports), so there is no need to rush to newer versions of Java. Improvements have been made since Java 11s' release that will likely benefit Dependency-Track in the future. The next LTS release Dependency-Track is aiming to support is Java 17, skipping Java 11 altogether.

**Comment by mendickxiao** [comment]
OK, I will suggest to update this document and tell people recommend for Java 8, not upgraded until the LTS support Java 17.
https://docs.dependencytrack.org/getting-started/deploy-war/
